homeassistant/components/media_player/samsungtv.py

```diff
@@ -256,6 +259,23 @@ def media_previous_track(self):
         """Send the previous track command."""
         self.send_key('KEY_REWIND')
 
+    async def async_play_media(self, media_type, media_id, **kwargs):
+        """Support changing a channel."""
+        if media_type != MEDIA_TYPE_CHANNEL:
+            _LOGGER.error('Unsupported media type')
+            return
+
+        # media_id should only be a channel number
+        try:
+            cv.positive_int(media_id)
+        except vol.Invalid:
+            _LOGGER.error('Media ID must be positive integer')
+            return
+
+        for digit in media_id:
+            self.hass.async_add_job(self.send_key, 'KEY_' + digit)
```

Comment:
I think we should `await` the returned task here. Otherwise it won't have finished before we sleep.